libwacom/libwacom-database.c

```diff
@@ -295,32 +419,65 @@ libwacom_parse_stylus_keyfile(WacomDeviceDatabase *db, const char *path)
 
 	groups = g_key_file_get_groups (keyfile, NULL);
 	for (i = 0; groups[i]; i++) {
-		WacomStylus *stylus;
-		WacomStylusId id;
+		WacomStylus *stylus, *other;
+		WacomStylusId id, lookup;
 		GError *error = NULL;
 		char *eraser_type, *type;
 		char **string_list;
+		char *alias_value;
 
 		if (!parse_stylus_id(groups[i], &id)) {
 			g_warning ("Failed to parse stylus ID '%s', ignoring entry", groups[i]);
 			continue;
 		}
+
 		stylus = g_new0 (WacomStylus, 1);
 		stylus->refcnt = 1;
 		stylus->id = id;
-		stylus->name = g_key_file_get_string(keyfile, groups[i], "Name", NULL);
-		stylus->group = g_key_file_get_string(keyfile, groups[i], "Group", NULL);
+
+		if (alias == IGNORE_ALIASES && g_key_file_get_string(keyfile, groups[i], "AliasOf", NULL) != NULL) {
+			continue;
+		}
+
+		if (alias == ONLY_ALIASES && g_key_file_get_string(keyfile, groups[i], "AliasOf", NULL) == NULL) {
+			continue;
+		}
+
+		if (alias == ONLY_ALIASES) {
+			alias_value = g_key_file_get_string(keyfile, groups[i], "AliasOf", NULL);
+
+			if (!create_stylus_id_from_aliased(groups[i], alias_value, &lookup)) {
```

Comment:
nickpick on the name:
```suggestion
			if (!create_stylus_id_from_alias(groups[i], alias_value, &lookup)) {
```